src/main/java/LinkedBinarySearchTree.java

```diff
@@ -0,0 +1,492 @@
+            }
+            returnElement = rightChild.getData();
+            parent.setRight(rightChild.getLeft());
+        }
+        size--;
+        return returnElement;
+    }
+
+    @Override
+    public T min() {
+        if (isEmpty()) {
+            throw new NoSuchElementException();
+        }
+        return min(root);
+    }
+
+    /**
+     * Helper method for enabling a recursive search for the minimum element
+     * in the binary search tree.
+     *
+     * @param current Current node being looked at
+     * @return The minimum element in the tree
+     */
+    private T min(Node<T> current) {
+        if (current.getLeft() == null) {
+            return current.getData();
+        } else {
+            return min(current.getLeft());
+        }
+    }
+
+    @Override
+    public T max() {
+        if (isEmpty()) {
+            throw new NoSuchElementException();
+        }
+        return max(root);
+    }
+
+    /**
+     * Helper method for enabling a recursive search for the maximum element
+     * in the binary search tree.
+     *
+     * @param current Current node being looked at
+     * @return The maximum element in the tree
+     */
+    private T max(Node<T> current) {
+        if (current.getRight() == null) {
+            return current.getData();
+        } else {
+            return max(current.getRight());
+        }
+    }
+
+    @Override
+    public boolean contains(T element) {
+        return binarySearch(element, root);
+    }
+
+    /**
+     * Helper method enabling a recursive binary search for a given element.
+     *
+     * @param element Element being searched for
+     * @param current Current binary search tree node being investigated
+     * @return True if the element is found, false otherwise
+     */
+    private boolean binarySearch(T element, Node<T> current) {
+        if (current == null) {
+            return false;
+        } else if (current.getData().equals(element)) {
+            return true;
+        } else {
+            if (current.getData().compareTo(element) > 0) {
+                return binarySearch(element, current.getLeft());
+            } else {
+                return binarySearch(element, current.getRight());
+            }
+        }
+    }
+
+    @Override
+    public int getCount(T element) {
+        if (isEmpty()) {
+            return 0;
+        }
+        return getCount(element, root);
+    }
+
+    /**
+     * Helper method for recursive count of elements in binary search tree.
+     * Remember, equal elements are added to the right subtree.
+     *
+     * @param element Element to be counted
+     * @param current Current node being investigated
+     * @return Number of times the element exists in the (sub)tree.
+     */
+    private int getCount(T element, Node<T> current) {
+        if (current == null) {
+            return 0;
+        }
+        if (current.getData().equals(element)) {
+            return 1 + getCount(element, current.getRight());
```

Comment:
This would be O(n). I figured when you added `getCount` to the interface, you'd be showing a log(n) solution here (two binary searches). Imo, remove or improve.

Reply:
I believe this is O(log(n)). The only catch is that if duplicates exist, we have to continue searching right. 

Unless you mean in case it gets degenerate, but I don't think I'll be doing any balancing, though, maybe am aside?

Reply:
Even assuming the tree is balanced so a search is always log(n), this algorithm is O(n).  
You could have a tree with a billion identical elements in it, you'd have to iterate the whole tree. Two binary searches would get you the result in `(2 * log(n) + 1)`, again, assuming the tree is balanced.

Reply:
I can't see an easy way to do this with a linked implementation. Since I am currently thinking of having them implement a `SortedBag` with this `LinkedBinarySearchTree`, having the `getCount` is great. If an aside on the `getCount` using binary search on an array, as discussed in #210, is included, it will give a chance to talk and compare about the time complexities. 

That said, as discussed with @twentylemon, this may really be pushing the limit of content for a CS102 course.